dcos/package.py

```diff
@@ -1354,39 +1351,55 @@ def package_versions(self):
         vs.reverse()
         return vs
 
-    def software_versions(self):
-        """Returns a mapping from the package version to the version of the
-        software described by the package.
+    def package_versions(self):
+        """Returns a mapping from the package revision to the package version.
 
-        :returns: Map from package versions to versions of the softwre.
+        :returns: Map from package revision to package version
         :rtype: dict
         """
 
-        software_package_map = collections.OrderedDict()
-        for v in self.package_versions():
+        package_version_map = collections.OrderedDict()
+        for v in self.package_revisions():
             pkg_json = self.package_json(v)
-            software_package_map[v] = pkg_json['version']
-        return software_package_map
+            package_version_map[v] = pkg_json['version']
+        return package_version_map
 
-    def latest_version(self):
-        """Returns the latest package version.
+    def latest_package_revision(self, package_version=None):
+        """Returns the most recent package revision, for a
+        given package version if specified.
 
-        :returns: The latest version of this package
-        :rtype: str
+        :param package_version: a given package version
+        :type package_version: str
+        :returns: package revision
+        :rtype: int
         """
 
-        pkg_versions = self.package_versions()
+        if package_version:
+            package_versions = self.package_versions()
+            # depends on package_versions() returning an OrderedDict
+            revision = next(
+                (pkg_ver for pkg_ver
+                    in reversed(package_versions)
+                    if package_versions[pkg_ver] == package_version), None)
+            if not revision:
+                raise DCOSException(
+                    "No package with version {} available".format(
```

Comment:
This is nitpicky, but it's slightly more clear to say "No package available with version [{}]"